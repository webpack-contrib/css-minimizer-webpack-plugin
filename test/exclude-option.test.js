import CssMinimizerPlugin from "../src/index";

import {
  compile,
  getCompiler,
  getErrors,
  getWarnings,
  readAssets,
} from "./helpers";

describe("exclude option", () => {
  let compiler;

  beforeEach(() => {
    compiler = getCompiler({
      entry: {
        excluded1: `${__dirname}/fixtures/excluded1.js`,
        excluded2: `${__dirname}/fixtures/excluded2.js`,
        entry: `${__dirname}/fixtures/entry.js`,
      },
    });
  });

  it("should match snapshot for a single RegExp value excluded1", async () => {
    new CssMinimizerPlugin({
      exclude: /excluded1/i,
    }).apply(compiler);

    const stats = await compile(compiler);

    expect(readAssets(compiler, stats, /\.css$/)).toMatchSnapshot("assets");
    expect(getErrors(stats)).toMatchSnapshot("errors");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
  });

  it("should match snapshot for a single String value excluded1", async () => {
    new CssMinimizerPlugin({
      exclude: "excluded1",
    }).apply(compiler);

    const stats = await compile(compiler);

    expect(readAssets(compiler, stats, /\.css$/)).toMatchSnapshot("assets");
    expect(getErrors(stats)).toMatchSnapshot("errors");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
  });

  it("should match snapshot for multiple RegExp values excluded1 and excluded2", async () => {
    new CssMinimizerPlugin({
      exclude: [/excluded1/i, /excluded2/i],
    }).apply(compiler);

    const stats = await compile(compiler);

    expect(readAssets(compiler, stats, /\.css$/)).toMatchSnapshot("assets");
    expect(getErrors(stats)).toMatchSnapshot("errors");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
  });

  it("should match snapshot for multiple String values excluded1 and excluded2", async () => {
    new CssMinimizerPlugin({
      exclude: ["excluded1", "excluded2"],
    }).apply(compiler);

    const stats = await compile(compiler);

    expect(readAssets(compiler, stats, /\.css$/)).toMatchSnapshot("assets");
    expect(getErrors(stats)).toMatchSnapshot("errors");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
  });
});
