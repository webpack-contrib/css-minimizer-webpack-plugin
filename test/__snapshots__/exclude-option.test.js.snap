// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`exclude option should match snapshot for a single RegExp value excluded1: assets 1`] = `
{
  "entry.css": "body{color:red}a{color:blue}",
  "excluded1.css": "a {
  text-align: center;
}

",
  "excluded2.css": "body{color:red;font-size:20px}a{color:blue}",
}
`;

exports[`exclude option should match snapshot for a single RegExp value excluded1: errors 1`] = `[]`;

exports[`exclude option should match snapshot for a single RegExp value excluded1: warnings 1`] = `[]`;

exports[`exclude option should match snapshot for a single String value excluded1: assets 1`] = `
{
  "entry.css": "body{color:red}a{color:blue}",
  "excluded1.css": "a {
  text-align: center;
}

",
  "excluded2.css": "body{color:red;font-size:20px}a{color:blue}",
}
`;

exports[`exclude option should match snapshot for a single String value excluded1: errors 1`] = `[]`;

exports[`exclude option should match snapshot for a single String value excluded1: warnings 1`] = `[]`;

exports[`exclude option should match snapshot for multiple RegExp values excluded1 and excluded2: assets 1`] = `
{
  "entry.css": "body{color:red}a{color:blue}",
  "excluded1.css": "a {
  text-align: center;
}

",
  "excluded2.css": "body {
  color: red;
}
body {
  color: red;
  font-size: 20px;
}
a {
  color: blue;
}

",
}
`;

exports[`exclude option should match snapshot for multiple RegExp values excluded1 and excluded2: errors 1`] = `[]`;

exports[`exclude option should match snapshot for multiple RegExp values excluded1 and excluded2: warnings 1`] = `[]`;

exports[`exclude option should match snapshot for multiple String values excluded1 and excluded2: assets 1`] = `
{
  "entry.css": "body{color:red}a{color:blue}",
  "excluded1.css": "a {
  text-align: center;
}

",
  "excluded2.css": "body {
  color: red;
}
body {
  color: red;
  font-size: 20px;
}
a {
  color: blue;
}

",
}
`;

exports[`exclude option should match snapshot for multiple String values excluded1 and excluded2: errors 1`] = `[]`;

exports[`exclude option should match snapshot for multiple String values excluded1 and excluded2: warnings 1`] = `[]`;
